### PR TITLE
use hunspell's add dictionary api to load known words

### LINKED ISSUE #14: add more common Go-isms to the known words set
Adding a few here as I've come across them. Some are rather generic to programming, others are somewhat specific to Go.

* builtins
* errored
* recurse
* config
* filesystem
* unexport
* codepoints
* tokenize
* lexing
* lexed
* backquote
* ascii
* unescaped
* unsetting
* html
* mutex
* vendored
* env (as in `go env`)
* funcs (as in plural of the `func` keyword)
* hacky
* uncomment
* v4 (as in major version, could be any vN)
* charset
* lossy
* hasher
* KiB (and MiB, etc?)

I realise I can add these to my local dictionary per-project, but I also assume you want to capture the common ones in the tool.

It's very likely that I'm not spelling some of these right, for example by omitting a dash like `un-setting`. Happy to be told to spell better :) But in those cases, it would be nice to be told what the right spelling is, because I don't.

### PR DESCRIPTION
**Pull request by kortschak**
@mvdan Please try this out, and suggest any words not in the list at #14 that you think would be good to have. The words from there that I have not included are misspellings.

Closes #14.

### CODE REVIEW

File: known_words.go [hunk 1]
```diff
@@ -5,33 +5,42 @@
...
+	"allocator/MS", "arg/MS", "async", "boolean/MS", "codec/MS", "endian",
+	"gcc/M", "glob/SDG", "globbing", "hostname/MS", "http/S", "localhost",
+	"mutex/MS", "NaN/S", "rpc/MS", "stderr/M", "stdin/M", "stdout/M", "symlink/MS",
+	"toolchain/MS", "ascii", "backquote/MS", "charset/MS", "codepoint/MS",
+	"comment/UMSGD", "config/MS", "env/MS", "error/DSM", "escaped/UDLMGS",
+	"export/UBSZGMDR", "filesystem/MS", "hacky", "hash/RAMDSG", "html/M",
+	"lex/GD", "lossy", "setting/U", "tokenize", "vendor/MSD",
+
+	// Units
+	"KiB/S", "MiB/S", "GiB/S", "TiB/S",
```
Comment (mvdan) [inline known_words.go:43]: might as well do KB/MB/GB/TB too

**Review by mvdan — APPROVED** [review]
SGTM code wise :)

What about "recurse" and "errored"? How am I meant to spell those?

File: known_words.go (hunk 1 above)
Comment (kortschak) [reply known_words.go:45]: Those will be ignored as all upper case.

**Comment by kortschak** [comment]
"error/DSM" matches "errored" and "recurse" is wrong, the correct word is "recurr".

**Comment by mvdan** [comment]
Are you sure that "errored" should be covered? I still see it:

```
/home/mvdan/src/sh/interp/handler.go:133:4: "errored" is misspelled in comment
	started, but errored - default to 1 if OS
	doesn't have exit statuses
```

**Comment by mvdan** [comment]
I also still get warned about "lexing", even though "lex" is here.

**Comment by mvdan** [comment]
Two more examples that still show up: "vendored" and "tokenized".

**Pull request merged**